mimesis/builtins/ru.py

```diff
@@ -90,16 +91,38 @@ def series_and_number(self) -> str:
     def snils(self) -> str:
         """Generate invalid ``SNILS``.
 
-        This method does not generate SNILS using algorithm and
-        it's mean that SNILS generated by this method can be invalid.
-
         :return: SNILS.
 
         :Example:
             451-952-540-41.
         """
-        mask = '###-###-###-##'
-        return self.random.custom_code(mask=mask)
+        snils = [self.random.randint(0, 9) for _ in range(9)]
+        while int(reduce(lambda x, y: x + y,
```

Comment:
Sorry, but this code looks to complicated for me.
Could we change how this loop works?

Things that are not clear for me:
1. what is `1001998`? if it is a constant let's give it a name
2. why do we have `snils = [self.random.randint(0, 9) for _ in range(9)]` twice?
3. can we please not use `reduce` and `map` with `lambda`s? In my personal opinion their usage is almost never required in `python`